src/System.Management.Automation/commands/utility/FormatAndOutput/out-console/OutConsole.cs

```diff
@@ -153,13 +151,15 @@ protected override void InternalDispose()
             }
 
             base.InternalDispose();
-            Host.UI.TranscribeOnly = _savedTranscribeOnly;
+            if (--Host.UI.TranscribeOnlyCount < 0)
```

Comment:
You are not checking if you actually increment `TranscribeOnlyCount` before. That lead to situation where every call to `Out-Default` reduce counter. Including ones without `-Transcript`. Or where begin block of `Out-Default` does not have opportunity to execute:
```powershell
& {begin{throw}} | Out-Default -Transcript
```